upbound_uxp/datadog_checks/upbound_uxp/check.py

```diff
@@ -0,0 +1,590 @@
+    'process_virtual_memory_bytes',
+    # Maximum amount of virtual memory available in bytes.
+    'process_virtual_memory_max_bytes',
+    # How many seconds has the longest running
+    # processor for workqueue been running.
+    'workqueue_longest_running_processor_seconds',
+    # How long in seconds an item stays
+    # in workqueue before being requested.
+    'workqueue_queue_duration_seconds_bucket',
+    'workqueue_queue_duration_seconds_count',
+    'workqueue_queue_duration_seconds_sum',
+    # Total number of retries handled by workqueue.
+    'workqueue_retries_total',
+    # How many seconds of work has been done
+    # that is in progress and hasn't been observed
+    # by work_duration. Large values indicate stuck
+    # threads. One can deduce the number of stuck
+    # threads by observing the rate at which this increases.
+    'workqueue_unfinished_work_seconds',
+]
+
+
+class UpboundUxpCheck(AgentCheck):
+    """
+    UpboundUXPCheck is a class to scrape Upbound's
+    Universal Crossplane and its Provider Kubernetes Pods
+
+    Minimal example configuration:
+
+    ```
+    init_config:
+
+    instances: |
+        [
+            {
+                "uxp_url": "/metrics",
+                "uxp_port": "8080",
+                # print useful info when verbose is true
+                "verbose": true,
+                # kubernetes namespace to check for
+                # uxp and provider containers
+                "namespace": "upbound-system",
+                # metric could be collected as often
+                # as min_collection_interval
+                "min_collection_interval": 30,
+                # default is unlimited
+                "metrics_limit": 60000,
+                # select from a predefined set of metrics:
+                # none, min, more, max
+                "metrics_default": "min",
+                # conf file flag to ignore otherwise
+                # superseding pod annotations
+                "metrics_ignore_pod_annotations": false
+                # prefix that will be inserted between uxp.
+                # and (mapped) metric name
+                "metrics_prefix":
+                # metrics examples below will be added to default,
+                # but pod annotations overwrite
+                "metrics": [
+                    {"go_goroutines": "company_prefix_go_goroutines"},
+                    # only one value needed without
+                    # name mapping requirements
+                    {"go_memstats_heap_alloc_bytes"},
+                ]
+            }
+        ]
+    ```
+    """
+
+    SERVICE_CHECK_CONNECT_NAME = 'upbound_uxp.can_connect'
+
+    # Prefix of every metric and service check that
+    # the upbound-uxp integration sends
+    __NAMESPACE__ = 'uxp'
+    DEFAULT_METRIC_LIMIT = 0
+
+    # Read agent configuration and initialize it
+    def __init__(self, name, init_config, instances):
+        print("DataDog Upbound Universal Crossplane Integration")
```

Comment:
Please remove `print` calls by the time this is ready to merge. Use functions under `self.log` (e.g. `self.log.debug`) if appropriate.